### PR TITLE
[ENH] Remove make_series from anomaly detection

### PR DESCRIPTION
**Pull request by TonyBagnall**
part of #2281 

anomaly detection tests were calling function to make a random numpy array. The function is legacy, and its not really required to create a random array

### CODE REVIEW

**Review by CodeLionX — CHANGES_REQUESTED** [review]
Thanks for taking on this task. I started to critisize usages of `make_series` in newer PRs.

File: aeon/anomaly_detection/tests/test_dwt_mlead.py [hunk 1]
```diff
@@ -6,20 +6,34 @@
...
+        ValueError, match="quantile_boundary_type must be 'percentile' or 'monte-carlo'"
+    ):
+        ad = DWT_MLEAD(quantile_boundary_type="Arsenal")
+        pred = ad.predict(series)
+    with pytest.raises(ValueError, match="epsilon must be in"):
+        ad = DWT_MLEAD(quantile_epsilon=-1.0)
+        pred = ad.predict(series)
+    with pytest.raises(ValueError):
+        ad = DWT_MLEAD(start_level=100)
+        pred = ad.predict(series)
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/anomaly_detection/tests/test_dwt_mlead.py:36]: sure, just adding coverage when I am in a module

File: aeon/anomaly_detection/tests/test_dwt_mlead.py [hunk 2]
```diff
@@ -6,20 +6,34 @@
...
 
 from aeon.anomaly_detection import DWT_MLEAD
-from aeon.testing.data_generation._legacy import make_series
 
 
 def test_dwt_mlead_output():
     """Test DWT_MLEAD output."""
-    series = make_series(n_timepoints=100, return_numpy=True, random_state=42)
+    rng = np.random.default_rng(seed=42)
+    series = rng.normal(size=(100,))
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/anomaly_detection/tests/test_dwt_mlead.py:14]: the problem that function creates a uniform distribution, If you use random uniformly distributed data then the tests of correct region all fail. We could remove those tests but that seemed beyond my remit to get rid of make series. Possibly resolve separately after this PR?

File: aeon/anomaly_detection/tests/test_iforest.py [hunk 3]
```diff
@@ -14,7 +14,8 @@
 )
 def test_iforest_default():
     """Test IsolationForest."""
-    series = make_series(n_timepoints=80, return_numpy=True, random_state=0)
+    rng = check_random_state(0)
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/anomaly_detection/tests/test_iforest.py:17]: I think check random state is better

**Comment by MatthewMiddlehurst** [comment]
This does not remove the function itself, but maybe that is best left to a separate PR.

File: aeon/anomaly_detection/tests/test_dwt_mlead.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (CodeLionX) [reply aeon/anomaly_detection/tests/test_dwt_mlead.py]: OK, now I get it. Your solution is also fine. Normal distribution makes more sense for anomaly detection 👍🏼

File: aeon/anomaly_detection/tests/test_stomp.py [hunk 4]
```diff
@@ -4,24 +4,29 @@
...
+    pred = ad.fit_predict(series)
     assert pred.shape == (80,)
     assert pred.dtype == np.float64
     assert 40 <= np.argmax(pred) <= 60
+    with pytest.raises(ValueError, match="The window size must be at least 1"):
+        ad = STOMP(window_size=0)
+        ad.fit_predict(series)
+    with pytest.raises(ValueError, match="The top `k` distances must be at least 1"):
+        ad = STOMP(k=0)
+        ad.fit_predict(series)
```
[the remark this answers is not in the document]
Comment (CodeLionX) [reply aeon/anomaly_detection/tests/test_stomp.py:32]: can also be removed

**Review by CodeLionX — CHANGES_REQUESTED** [review]

**Review by CodeLionX — APPROVED** [review]

**Pull request merged**